components/xpath/src/lib.rs

```diff
@@ -56,6 +56,7 @@ pub trait Node: Eq + Clone + Debug {
     fn as_attribute(&self) -> Option<Self::Attribute>;
     fn as_element(&self) -> Option<Self::Element>;
     fn lookup_namespace_uri(&self, uri: Option<&str>) -> Option<String>;
+    fn get_root_node(&self) -> Self;
```

Comment:
I added a new trait method here because while we could traverse the node's ancestors, it is often possible to get a node's root node in `O(1)` if it is in a shadow tree or a document tree.